### PR TITLE
feat(AUTH-CPC-974): auth create user with inventory manager role

### PR DESCRIPTION
**Pull request by DylanBrass**
JIRA: [link to jira ticket](https://champlainsaintlambert.atlassian.net/browse/CPC-974?atlOrigin=eyJpIjoiZGNlNjk3ZjRjYWNmNGU3Nzk4NWI4YWU5YzhmZTk2NGIiLCJwIjoiaiJ9)
## Context:
This ticket allows the creation of a user that is specialized to handle the inventories. This user has the inventory manager role and can now be used to secure endpoint with the custom annotations.
## Changes
- Added new page to create inventory manager in admin panel
- Added new endpoint to create an inventory manager
- Fixed user details, to be continued by teammate 
- Added tests to auth service client to cover previously uncovered functions
### Before : 

image

### After : 

image

image

### CODE REVIEW

File: api-gateway/src/main/java/com/petclinic/bffapigateway/exceptions/GenericHttpException.java [hunk 1]
```diff
@@ -12,6 +13,7 @@
  */
 
 @Data
+@EqualsAndHashCode(callSuper = false)
```
Comment (RichardDanon) [inline api-gateway/src/main/java/com/petclinic/bffapigateway/exceptions/GenericHttpException.java:16]: What is this doing?

File: vet-service/src/main/java/com/petclinic/vet/dataaccesslayer/ratings/RatingRepository.java [hunk 2]
```diff
@@ -11,5 +11,7 @@ public interface RatingRepository extends ReactiveMongoRepository<Rating, String
     Mono<Long> countAllByVetId(String vetId);
     Mono<Rating> findByVetIdAndRatingId(String vetId, String ratingId);
     Mono<Rating> findByRatingId(String ratingId);
+
+
```
Comment (RichardDanon) [inline vet-service/src/main/java/com/petclinic/vet/dataaccesslayer/ratings/RatingRepository.java:15]: "Nitpick"

You don't need to add spaces if they don't change anything (you probably just forgot to remove them).

**Review by RichardDanon — APPROVED** [review]
GreenLighted everything looks good

File: api-gateway/src/main/java/com/petclinic/bffapigateway/exceptions/GenericHttpException.java (hunk 1 above)
Comment (DylanBrass) [reply api-gateway/src/main/java/com/petclinic/bffapigateway/exceptions/GenericHttpException.java:16]: This  to stop the @Data to calling the weird super constructor. This is a funny quirk of lomdok !

**Review by AmirHamzahMirza — APPROVED** [review]
You have a lot of miscellaneous added lines that say you changed files, I think you should fix those but honestly that's just a nitpick and there's no real issues.

**Pull request merged**